int32 m_num
---
float32 x
float32 y
float32 theta
string action
string color